packages/terra-action-header/src/ActionHeader.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from 'terra-button';
+import IconLeft from 'terra-icon/lib/icon/IconLeft';
+import IconClose from 'terra-icon/lib/icon/IconClose';
+import ResponsiveElement from 'terra-responsive-element';
+import 'terra-base/lib/baseStyles';
+import ActionHeaderContainer from './_ActionHeaderContainer';
+
+const propTypes = {
+  /**
+   * Text to be displayed as the title in the header bar.
+   */
+  title: PropTypes.string,
+  /**
+   * Optionally sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`. Default `level=1`.
+   */
+  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
+  /**
+   * Callback function for when the close button is clicked.
+   * On small viewports a back button will be displayed instead of a close button when a separate onBack callback is not set.
+   */
+  onClose: PropTypes.func,
+  /**
+   * Callback function for when the back button is clicked. The back button will not display if this is not set.
+   */
+  onBack: PropTypes.func,
+  /**
+   * A Boolean indicating if close button should be displayed on small viewports when separate onBack callback is not set.
+   */
+  keepCloseButton: PropTypes.bool,
+  /**
+   * Child element to be displayed on the right end of the header.
+   */
+  children: PropTypes.element,
+};
+
+const defaultProps = {
+  title: undefined,
+  level: 1,
+  onClose: undefined,
+  onBack: undefined,
+  keepCloseButton: false,
+  children: undefined,
+};
+
+const contextTypes = {
```

Comment:
Once `terra-base` has been updated to expose `injectIntl`, we should use that instead of context for translations.

Reply:
Is that a part of #1350?

Reply:
No this is a PR coming soon by @tbiethman

Reply:
I agree with @yuderekyu, though the work to update this code to use `injectIntl` can be completed in a subsequent PR.

Reply:
FYI: #1379